### PR TITLE
Fix bad use of channels last kernel in sync batch norm backward

### LINKED ISSUE #64039: SyncBatchNorm + BatchNorm2d produces incorrect gradients
## 🐛 Bug

Under certain conditions using `SyncBatchNorm` + `BatchNorm2d` produces incorrect gradients, this is a regression on torch 1.9.1 where the behavior is correct.

## To Reproduce

```py
import os
import torch
import torch.distributed as dist
import torch.multiprocessing as mp

from torch.nn.parallel import DistributedDataParallel as DDP


class BatchNorm1dModel(torch.nn.Module):

    def __init__(self, feat_dim):
        super().__init__()
        self.bnorm = torch.nn.BatchNorm1d(feat_dim)
        self.linear1 = torch.nn.Linear(feat_dim, feat_dim)
        self.linear2 = torch.nn.Linear(feat_dim, feat_dim)

    def forward(self, x):
        x = self.linear1(x)
        x = self.bnorm(x.transpose(1,2)).transpose(1,2)
        x = self.linear2(x)
        return x


class BatchNorm2dModel(torch.nn.Module):

    def __init__(self, feat_dim):
        super().__init__()
        self.bnorm = torch.nn.BatchNorm2d(feat_dim)
        self.linear1 = torch.nn.Linear(feat_dim, feat_dim)
        self.linear2 = torch.nn.Linear(feat_dim, feat_dim)

    def forward(self, x):
        x = self.linear1(x)
        x = x.unsqueeze(-1)
        x = self.bnorm(x.transpose(1,2)).transpose(1,2)
        x = x.squeeze(-1)
        x = self.linear2(x)
        return x


def main(rank):
    # iniialize the process group
    os.environ["MASTER_ADDR"] = "localhost"
    os.environ["MASTER_PORT"] = "12355"
    dist.init_process_group(backend="gloo", rank=rank, world_size=2)

    torch.manual_seed(0)
    model_1d = BatchNorm1dModel(3).to(rank).double()

    torch.manual_seed(0)
    model_2d = BatchNorm2dModel(3).to(rank).double()

    inputs = torch.randn(2, 2, 3, requires_grad=True).double()

    for model in [model_1d, model_2d]:
        model = torch.nn.SyncBatchNorm.convert_sync_batchnorm(model)
        model = DDP(model, device_ids=[rank])

        outputs = model(inputs)
        outputs.sum().backward()
        if rank == 0:
            print("OUTPUT\n", outputs)
            print("LINEAR1 GRAD\n", model.module.linear1.weight.grad)
            print("LINEAR2 GRAD\n", model.module.linear2.weight.grad, end="\n\n")

        torch.autograd.gradcheck(model, (inputs,))


if __name__ == "__main__":
    mp.spawn(main, nprocs=2, join=True)
```
Produces the following output
![image](https://user-images.githubusercontent.com/17232877/131016670-44e65859-34d5-42ff-aab3-6f6413d81fe2.png)

and raises an error from gradcheck
![image](https://user-images.githubusercontent.com/17232877/131016603-6d193166-80fd-4334-8b5b-13d7fafec3f3.png)


## Expected behavior

The outputs and the gradients for the second linear layer are consistent between `BatchNorm2d` and `BatchNormd1d` models, but the gradients for the first linear layer that must pass through batchnorm layer are incorrect, as confirmed by `torch.autograd.gradcheck`. This behavior does not happen on torch 1.9.1.

## Environment

```
Collecting environment information...                                                                                                                                     
PyTorch version: 1.10.0a0+gitc545b09                                                                                                                                      
Is debug build: False                                                                                                                                                     
CUDA used to build PyTorch: 11.2                                                                                                                                          
ROCM used to build PyTorch: N/A                                                                                                                                                                                                                                                                                                                     OS: Ubuntu 18.04.5 LTS (x86_64)                                                                                                                                           
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0                                                                                                                          Clang version: Could not collect                                                                                                                                          
CMake version: version 3.21.1                                                                                                                                             
Libc version: glibc-2.27                                                                                                                                                                                                                                                                                                                            Python version: 3.8.11 (default, Jul  3 2021, 17:53:42)  [GCC 7.5.0] (64-bit runtime)                                                                                     Python platform: Linux-5.4.0-80-generic-x86_64-with-glibc2.27                                                                                                             Is CUDA available: True                                                                                                                                                   
CUDA runtime version: 11.2.152                                                                                                                                            
GPU models and configuration:
GPU 0: GeForce RTX 2080 Ti
GPU 1: GeForce RTX 2080 Ti
GPU 2: GeForce RTX 2080 Ti
GPU 3: GeForce RTX 2080 Ti
GPU 4: GeForce RTX 2080 Ti
GPU 5: GeForce RTX 2080 Ti
GPU 6: GeForce RTX 2080 Ti
GPU 7: GeForce RTX 2080 Ti
GPU 8: GeForce RTX 2080 Ti
GPU 9: GeForce RTX 2080 Ti

Nvidia driver version: 460.32.03
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.1.1
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] pytorch-memlab==0.2.3
[pip3] torch==1.10.0a0+gitc545b09
[pip3] torchaudio==0.10.0a0+3bb5feb
[pip3] torchpq==0.1.8.3
[pip3] torchsummary==1.5.1
```

cc @albanD @mruberry @jbschlosser

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #64039

There are two distinct problems here.
1. If `grad_output` is channels last but not input, then input would be read as-if it were channels last. So reading the wrong values.
2. `use_channels_last_kernels` doesn't guarunte that `suggest_memory_format` will actually return channels last, so use `empty_like` instead so the strides always match.

### CODE REVIEW

**Review by ngimel — APPROVED** [review]
Thanks for a quick fix, @peterbell10!

**Review by ngimel — APPROVED** [review]
